zookeeper-server/src/test/java/org/apache/zookeeper/test/ZooKeeperTestClient.java

```diff
@@ -18,450 +18,435 @@
+        }
 
-    if (!children1.equals(c2)) {
-        Assert.fail("children lists from getChildren()/getChildren2() do not match");
-    }
+        if (!stat.equals(stat)) {
+            Assert.fail("stats from exists()/getChildren2() do not match");
+        }
 
-    if (!stat.equals(stat)) {
-        Assert.fail("stats from exists()/getChildren2() do not match");
+        if (children1.size() == 0) {
+            zk.delete(nodeName, -1);
+            return;
+        }
+        for (String n : children1) {
+            deleteZKDir(zk, n);
+        }
     }
 
-    if (children1.size() == 0) {
-      zk.delete(nodeName, -1);
-      return;
-    }
-    for (String n : children1) {
-      deleteZKDir(zk, n);
-    }
-  }
-
-  private void checkRoot() throws IOException,
-      InterruptedException {
-    ZooKeeper zk = new ZooKeeper(hostPort, 10000, this);
-
-    try {
-      zk.create(dirOnZK, null, Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-    } catch (KeeperException.NodeExistsException ke) {
-        // expected, sort of
-    } catch (KeeperException ke) {
-        Assert.fail("Unexpected exception code for create " + dirOnZK + ": "
-            + ke.getMessage());
-    }
+    private void checkRoot() throws IOException, InterruptedException {
+        ZooKeeper zk = new ZooKeeper(hostPort, 10000, this);
 
-    try {
-      zk.create(testDirOnZK, null, Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-    } catch (KeeperException.NodeExistsException ke) {
-        // expected, sort of
-    } catch (KeeperException ke) {
-        Assert.fail("Unexpected exception code for create " + testDirOnZK + ": "
-            + ke.getMessage());
-    }
+        try {
+            zk.create(dirOnZK, null, Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+        } catch (KeeperException.NodeExistsException ke) {
+            // expected, sort of
+        } catch (KeeperException ke) {
+            Assert.fail("Unexpected exception code for create " + dirOnZK + ": " + ke.getMessage());
+        }
 
-    zk.close();
-  }
-
-  private void enode_test_1() throws IOException,
-          InterruptedException, KeeperException {
-    checkRoot();
-    String parentName = testDirOnZK;
-    String nodeName = parentName + "/enode_abc";
-    ZooKeeper zk = new ZooKeeper(hostPort, 10000, this);
-
-    Stat stat = zk.exists(parentName, false);
-    if (stat == null) {
-      try {
-        zk.create(parentName, null, Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
-      } catch (KeeperException ke) {
-        Assert.fail("Creating node " + parentName + ke.getMessage());
-      }
-    }
+        try {
+            zk.create(testDirOnZK, null, Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+        } catch (KeeperException.NodeExistsException ke) {
+            // expected, sort of
+        } catch (KeeperException ke) {
+            Assert.fail("Unexpected exception code for create " + testDirOnZK + ": " + ke.getMessage());
```

Comment:
I noticed only now this fact: don't we have a rule to forbid the usage of "Assert.xxxx" and force the usage of import static "org.junit.Assert.fail" ?

Reply:
We can add a forbidden pattern `import org.junit.Assert;`

Reply:
the rule should be like 

```xml
    <module name="RegexpSingleline">
        <!-- Checks that TODOs don't have stuff in parenthesis, e.g., username. -->
        <property name="format" value="import org\.junit\.Assert;" />
        <property name="message" value="Always import static org.junit.Assert.[static-method]." />
        <property name="severity" value="error" />
    </module>
```

but I would like do it in another pass since it would affect on other modules.

Reply:
Not needed for me.
Let's keep it as it is now.
Thanks

Reply:
Hmmm I have done it with a script remove `import org.junit.Assert;` and replace ` Assert.` with ` `. And then add a redundant `import static` headers which is followed by an imports clean up automate phase.

I do it inside zookeeper-server scope and it should not be hard to review.(just import -> import static_